tests/test_docker_publish_workflow.py

```
@@ -0,0 +1,40 @@
+    def test_workflow_configures_qemu_before_buildx_for_multi_arch_images(self) -> None:
+        qemu_marker = "uses: docker/setup-qemu-action@"
+        buildx_marker = "uses: docker/setup-buildx-action@"
+
+        qemu_index = next(
+            index for index, line in enumerate(self.lines) if qemu_marker in line
+        )
+        buildx_index = next(
+            index for index, line in enumerate(self.lines) if buildx_marker in line
+        )
+
+        self.assertIn(qemu_marker, self.content)
```

Ruff (0.15.5)

[warning] Use a regular `assert` instead of unittest-style `assertIn`

Replace `assertIn(...)` with `assert ...`

(PT009)